### PR TITLE
virts-1470 planning service tests

### PR DESCRIPTION
**Pull request by elegantmoose**
n/a

### CODE REVIEW

**Comment by elegantmoose** [comment]
Please review @christophert .

Build errors seem to be from third party lib deprecation warning.

**Comment by elegantmoose** [comment]
NOTE: If https://github.com/mitre/caldera/pull/1849 gets merged first, need to tweak tests as that will break them (superficially, not logically).

**Review by christophert — APPROVED** [review]

File: tests/services/test_planning_svc.py [hunk 1]
```diff
@@ -30,6 +30,157 @@ def setup_planning_test(loop, ability, agent, operation, data_svc, init_base_wor
...
+
+            async def three(self):
+                self.calls.append('three')
+                self.next_bucket = None  # stopping execution here
+
+            async def four(self):
+                self.calls.append('four')
+                self.next_bucket = None
+
+        # case 1 - let planner run until it stops itself after bucket 'three'
```
Comment (wbooth) [inline tests/services/test_planning_svc.py:159]: can you break each of these cases into separate tests

File: tests/services/test_planning_svc.py [hunk 2]
```diff
@@ -30,6 +30,157 @@ def setup_planning_test(loop, ability, agent, operation, data_svc, init_base_wor
...
+        # case 3 - operaton finished, planner stop condition not met
+        operation.state = operation.states['FINISHED']
+        p.stopping_condition_met = False
+        assert loop.run_until_complete(planning_svc._stop_bucket_exhaustion(p, operation, True)) is True
+        # case 4 - operation finished, planner stop condition met
+        p.stopping_condition_met = True
+        assert loop.run_until_complete(planning_svc._stop_bucket_exhaustion(p, operation, True)) is True
+        # case 5 - case 2 with condition stop off
+        operation.state = operation.states['RUNNING']
+        assert loop.run_until_complete(planning_svc._stop_bucket_exhaustion(p, operation, False)) is False
```
Comment (wbooth) [inline tests/services/test_planning_svc.py:128]: this is a good candidate for a fixture + parameterization

**Review by wbooth — DISMISSED** [review]
see comments

**Comment by elegantmoose** [comment]
Changes addressed. @wbooth

**Pull request merged**